src/App.js

```diff
@@ -16,53 +16,53 @@ function App() {
   const [isLoading, setIsLoading] = useState(true);
   const [requestError, setRequestError] = useState(false);
 
-
-//  random number between 1 and 154
+  //  random number between 1 and 154
   function randomNumber() {
     // min and max included
     return Math.floor(Math.random() * 154 + 1);
   }
 
-
-// make api request on mount
+  // make api request on mount
   useEffect(() => {
     fetch("https://poetrydb.org/author,title/Shakespeare;Sonnet")
       .then((response) => {
-       setRequestError(!response.ok);
-       if (response.ok) {
-         return response.json();
-       } else {
-         throw Error('Response status: ' + response.status)
-       }
-      }).then((result) => {
+        setRequestError(!response.ok);
+        if (response.ok) {
+          return response.json();
+        } else {
+          throw Error("Response status: " + response.status);
+        }
+      })
+      .then((result) => {
         result.forEach(function (element, index) {
-          element['number'] = index + 1
-        })
-        setApiData(result)
-        setNumber(randomNumber())
-        setIsLoading(false)
-        }).catch((error) => {
-          console.log(error)
+          element["number"] = index + 1;
+        });
+        setApiData(result);
+        setNumber(randomNumber());
+        setIsLoading(false);
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  }, []);
+
+  // select different poem every time number changes
+  useEffect(() => {
+    if (number > 0) {
+      setPoem(
+        apiData.find((poem) => {
+          return poem.number === Number(number);
         })
```

Comment:
```javascript
apiData.find((poem) => {
  return poem.number === Number(number);
})
```

There is a much simpler way to do this no?

Reply:
Initially, I was setting the poem by index... setPoem(apiData[number-1])... this is obviously simpler. When I refactored, I thought it might be 'purer' not have to mess around with the -1. This is a case of me overthinking it slightly. My first instinct was to use the index. I'll redo.